### PR TITLE
`<flat_map>` Fix compile errors

### PR DESCRIPTION
**Pull request by duckdoom5**
As mentioned in a [comment](https://github.com/microsoft/STL/commit/00c23249b9580512990722e3a0494fe61866c3fe#commitcomment-137907936), I've been testing out the `flat_map` implementation in my personal project but after the last commit I'm getting the following error:
`flat_map.h(1196,44): '_Key_equal': function was not declared in the template definition context and can be found only via argument-dependent lookup in the instantiation context`

Previous version compiled fine though.

It seems the `_Try_emplace` function here depends on the private `_Key_equal`. I moved that back to the base implementation similar to `_Erase_if`.

I've also moved `_Insert_exact` to the protected section as the `flat_multimap::_Emplace_key_mapped()` uses that function.

### CODE REVIEW

File: stl/inc/flat_map [hunk 1]
```diff
@@ -1328,6 +1329,7 @@ public:
 
 private:
     using _MyBase::_Erase_if;
+    using _MyBase::_Insert_exact
```
Comment (frederick-vs-ja) [inline stl/inc/flat_map:1332]: Clang-format mis-suggested. The real error is the missing of a `;`.
```suggestion
    using _MyBase::_Insert_exact;
```

Comment (duckdoom5) [reply stl/inc/flat_map]: Haha, yeah I noticed. Fixed now :)

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for fixing this bug! :heart_eyes_cat: I added test coverage for `try_emplace()` (after verifying that it caused the error you described); the test is still a work in progress but we should make sure that this fix stays fixed. I also had to change the test to be "strict" since `/permissive` mode rejects `try_emplace()` here.

**Comment by StephanTLavavej** [comment]
Congratulations on your first microsoft/STL commit! :tada: :rocket: :smile_cat:

**Comment by duckdoom5** [comment]
> Congratulations on your first microsoft/STL commit! :tada: :rocket: :smile_cat:

Thanks! 🎉😸

**Pull request merged**